src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,201 @@
+            userInput.clear();
+        });
+
+        userInput.setOnAction((event) -> {
+            dialogContainer.getChildren().add(getDialogLabel(userInput.getText()));
+            userInput.clear();
+        });
+
+        sendButton.setOnMouseClicked((event) -> {
+            handleUserInput();
+        });
+
+        userInput.setOnAction((event) -> {
+            handleUserInput();
+        });
+    }
+
+    /**
+     * Iteration 1:
+     * Creates a label with the specified text and adds it to the dialog container.
+     * @param text String containing text to add
+     * @return a label with the specified text that has word wrap enabled.
+     */
+    private Label getDialogLabel(String text) {
+        // You will need to import `javafx.scene.control.Label`.
+        Label textToAdd = new Label(text);
+        textToAdd.setWrapText(true);
+        return textToAdd;
+    }
+
+    /**
+     * Iteration 2:
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
+     * the dialog container. Clears the user input after processing.
+     */
+    private void handleUserInput() {
+        String input = userInput.getText();
+        String dukeProcessedText = getResponse(input);
+        dialogContainer.getChildren().addAll(
+                DialogBox.getUserDialog(userInput.getText(), userImage),
+                DialogBox.getDukeDialog(dukeProcessedText, dukeImage)
+        );
+        userInput.clear();
+    }
+
+    public String getResponse(String input) {
+        String dukeProcessedText;
+        String[] parsedText = Parser.parse(input, tasks);
+        switch(parsedText[0]) {
+            case "mark" :
+                dukeProcessedText = tasks.markTask(Integer.parseInt(parsedText[1]));
+                break;
+            case "unmark" :
+                dukeProcessedText = tasks.unmarkTask(Integer.parseInt(parsedText[1]));
+                break;
+            case "delete" :
+                dukeProcessedText = tasks.deleteTask(Integer.parseInt(parsedText[1]));
+                break;
+            case "find" :
+                try {
+                    dukeProcessedText = tasks.find(parsedText[1]);
+                } catch (DukeException e) {
+                    dukeProcessedText = e.getMessage();
+                }
+                break;
+            case "list" :
+                dukeProcessedText = tasks.getAll();
+                break;
+            case "exception" :
+                dukeProcessedText = parsedText[1];
+                break;
+            case "todo" :
+            case "deadline" :
+            case "event" :
+                try {
+                    dukeProcessedText = tasks.addTask(parsedText);
+                } catch (DukeException e) {
+                    dukeProcessedText = e.getMessage();
+                }
+                break;
+            default:
+                dukeProcessedText = "Invalid input";
+                break;
```

Comment:
There should be no indentation for case clauses as according to the Java coding standard. You can change the default arrangement of your IDE :).